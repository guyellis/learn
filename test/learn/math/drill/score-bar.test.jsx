const React = require('react');
const renderer = require('react-test-renderer');
const ScoreBar = require('../../../../src/learn/math/drill/score-bar');

function mockDates(times) {
  Date.now = jest.fn();
  times.forEach((time) => {
    const mockDate = time.date + time.dateMockDelta;
    Date.now.mockImplementationOnce(() => mockDate);
  });
}

describe('ScoreBar', () => {
  const nowFn = Date.now;
  afterAll(() => {
    Date.now = nowFn;
  });

  test('should not show if showScoreBar is false', () => {
    const times = [];

    const component = renderer.create(<ScoreBar
      showScoreBar={false}
      times={times}
    />);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should not show if times is empty', () => {
    const times = [];

    const component = renderer.create(<ScoreBar
      showScoreBar
      times={times}
    />);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should show 1 column with Gold Badge', () => {
    const times = [{
      date: 1506395542848, // 9/25/2017 8:12pm PST
      timePerQuestion: 1.5,
      dateMockDelta: 1000 * 60 * 4,
    }];

    mockDates(times);

    const component = renderer.create(<ScoreBar
      showScoreBar
      times={times}
    />);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should show 1 column with below Blue Badge', () => {
    const times = [{
      date: 1506395542848, // 9/25/2017 8:12pm PST
      timePerQuestion: 15.2,
      dateMockDelta: 1000 * 60 * 4,
    }];

    mockDates(times);

    const component = renderer.create(<ScoreBar
      showScoreBar
      times={times}
    />);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should show 5 columns with mixed Badges', () => {
    const times = [{
      date: 1506395542848, // 9/25/2017 8:12pm PST
      timePerQuestion: 15.2,
      dateMockDelta: 1000 * 60 * 4,
    }, {
      date: 1506395642848, // A bit later
      timePerQuestion: 10.2,
      dateMockDelta: 1000 * 60 * 2,
    }, {
      date: 1506395742848, // And later
      timePerQuestion: 5.2,
      dateMockDelta: 1000 * 3,
    }];

    mockDates(times);

    const component = renderer.create(<ScoreBar
      showScoreBar
      times={times}
    />);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });
});
