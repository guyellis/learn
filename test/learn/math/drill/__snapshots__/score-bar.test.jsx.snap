// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBar should not show if showScoreBar is false 1`] = `null`;

exports[`ScoreBar should not show if times is empty 1`] = `null`;

exports[`ScoreBar should show 1 column with Gold Badge 1`] = `
<div
  style={Object {}}
>
  <div
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        4 minutes ago
      </span>
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        Badges
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "height": "300px",
        "width": "100%",
      }
    }
  >
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "300px",
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "height": "20.0%",
          }
        }
      >
        
      </span>
      <span
        style={
          Object {
            "backgroundColor": "#fbc02d",
            "height": "10%",
          }
        }
      >
        YOU⇾
      </span>
      <span
        style={
          Object {
            "height": "70.0%",
          }
        }
      >
        
      </span>
    </span>
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "300px",
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fbc02d",
            "height": "33%",
          }
        }
      >
        Gold
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#9e9e9e",
            "height": "17%",
          }
        }
      >
        Silver
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#8d6e63",
            "height": "17%",
          }
        }
      >
        Bronze
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "33%",
          }
        }
      >
        Blue
      </div>
    </span>
  </div>
</div>
`;

exports[`ScoreBar should show 1 column with below Blue Badge 1`] = `
<div
  style={Object {}}
>
  <div
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        4 minutes ago
      </span>
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        Badges
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "height": "400px",
        "width": "100%",
      }
    }
  >
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "400px",
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "height": "95.0%",
          }
        }
      >
        
      </span>
      <span
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "10%",
          }
        }
      >
        YOU↓
      </span>
      <span
        style={
          Object {
            "height": "0.0%",
          }
        }
      >
        
      </span>
    </span>
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "400px",
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fbc02d",
            "height": "25%",
          }
        }
      >
        Gold
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#9e9e9e",
            "height": "13%",
          }
        }
      >
        Silver
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#8d6e63",
            "height": "13%",
          }
        }
      >
        Bronze
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "50%",
          }
        }
      >
        Blue
      </div>
    </span>
  </div>
</div>
`;

exports[`ScoreBar should show 5 columns with mixed Badges 1`] = `
<div
  style={Object {}}
>
  <div
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        4 minutes ago
      </span>
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        2 minutes ago
      </span>
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        a few seconds ago
      </span>
      <span
        style={
          Object {
            "flexGrow": "1",
          }
        }
      >
        Badges
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "height": "400px",
        "width": "100%",
      }
    }
  >
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "400px",
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "height": "95.0%",
          }
        }
      >
        
      </span>
      <span
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "10%",
          }
        }
      >
        YOU↓
      </span>
      <span
        style={
          Object {
            "height": "0.0%",
          }
        }
      >
        
      </span>
    </span>
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "400px",
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "height": "95.0%",
          }
        }
      >
        
      </span>
      <span
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "10%",
          }
        }
      >
        YOU↓
      </span>
      <span
        style={
          Object {
            "height": "0.0%",
          }
        }
      >
        
      </span>
    </span>
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "400px",
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "height": "60.0%",
          }
        }
      >
        
      </span>
      <span
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "10%",
          }
        }
      >
        YOU⇾
      </span>
      <span
        style={
          Object {
            "height": "30.0%",
          }
        }
      >
        
      </span>
    </span>
    <span
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": "1",
          "fontSize": "large",
          "height": "400px",
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fbc02d",
            "height": "25%",
          }
        }
      >
        Gold
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#9e9e9e",
            "height": "13%",
          }
        }
      >
        Silver
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#8d6e63",
            "height": "13%",
          }
        }
      >
        Bronze
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#81d4fa",
            "height": "50%",
          }
        }
      >
        Blue
      </div>
    </span>
  </div>
</div>
`;
